src/modules/collection/sagas.ts

```diff
@@ -255,7 +255,9 @@ export function* collectionSaga(builder: BuilderAPI, catalyst: CatalystClient) {
     try {
       yield call(() => builder.deleteCollection(collection.id))
       yield put(deleteCollectionSuccess(collection))
-      const collectionIdInUriParam: string = yield select(getCollectionId)
+
+      const collectionIdSelector = isTPCollection(collection) ? getThirdPartyCollectionId : getCollectionId
```

Comment:
What if we embed this logic in the selector? We can just use the `getCollectionId` and add the `thirdPartyCollectionDetail` to the match selector that we have:
```
const collectionIdMatchSelector = createMatchSelector<
  RootState,
  {
    collectionId: string
  }
>([locations.collectionDetail(), locations.thirdPartyCollectionDetail()])
```
I believe that if we add it there, we can re-use the same selector for both types of collection :) 

Just a suggestion, I'm ok with these changes too!

Reply:
Woo I love this. Looks much better.